importlib_metadata/__init__.py

```diff
@@ -308,7 +359,8 @@ def version(self):
 
     @property
     def entry_points(self):
-        return list(EntryPoint._from_text_for(self.read_text('entry_points.txt'), self))
+        eps = EntryPoint._from_text_for(self.read_text('entry_points.txt'), self)
+        return GroupedEntryPoints(eps)
```

Comment:
I believe this changes the type from `list` to a mapping-like -- this doesn't seem correct?

Reply:
It changes from a list to a GroupedEntryPoints, which is a tuple subclass with `__getitem__` for easy access by group (and no access by index). It's a little magical, but well within the norms of OO programming.